runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeItemPanel.java

```diff
@@ -131,27 +134,41 @@ public void mouseReleased(MouseEvent e)
 			gePriceLabel.setText("N/A");
 		}
 		gePriceLabel.setForeground(ColorScheme.GRAND_EXCHANGE_PRICE);
-		rightPanel.add(gePriceLabel);
-
-		JPanel alchAndLimitPanel = new JPanel(new BorderLayout());
-		panels.add(alchAndLimitPanel);
-		alchAndLimitPanel.setBackground(background);
-
-		// Alch price
-		JLabel haPriceLabel = new JLabel();
-		haPriceLabel.setText(StackFormatter.formatNumber(haPrice.intValue()) + " alch");
-		haPriceLabel.setForeground(ColorScheme.GRAND_EXCHANGE_ALCH);
-		alchAndLimitPanel.add(haPriceLabel, BorderLayout.WEST);
+		gePriceAndLimitPanel.add(gePriceLabel, BorderLayout.WEST);
 
 		// GE Limit
 		JLabel geLimitLabel = new JLabel();
 		String limitLabelText = geItemLimit == 0 ? "" : "Limit " + StackFormatter.formatNumber(geItemLimit);
 		geLimitLabel.setText(limitLabelText);
 		geLimitLabel.setForeground(ColorScheme.GRAND_EXCHANGE_LIMIT);
 		geLimitLabel.setBorder(new CompoundBorder(geLimitLabel.getBorder(), new EmptyBorder(0, 0, 0, 7)));
-		alchAndLimitPanel.add(geLimitLabel, BorderLayout.EAST);
+		gePriceAndLimitPanel.add(geLimitLabel, BorderLayout.EAST);
 
-		rightPanel.add(alchAndLimitPanel);
+		rightPanel.add(gePriceAndLimitPanel);
+
+		// Alch and Profit
+		JPanel alchAndProfitPanel = new JPanel(new BorderLayout());
+		panels.add(alchAndProfitPanel);
+		alchAndProfitPanel.setBackground(background);
+
+		// Alch price
+		JLabel haPriceLabel = new JLabel();
+		String haPriceLabelText = StackFormatter.formatNumber(haPrice.intValue()) + " alch";
+		haPriceLabel.setText(haPriceLabelText);
+		haPriceLabel.setForeground(ColorScheme.GRAND_EXCHANGE_ALCH);
+		alchAndProfitPanel.add(haPriceLabel, BorderLayout.WEST);
+
+		// Profit Margin
+		JLabel profitLabel = new JLabel();
```

Comment:
I'd personally set all the variables before getting into the UI part, like:

int profitMargin...
Color color...
String string...

JLabel profitLabel...
profitLabel.setText
profitLabel.setForeground

alchAndProfitPanel.add(..)
rightPanel.add(..)

Just so it's clearer, should make it more readable, but that's just my style preference.

Reply:
That makes sense to me. Also, having changed it, it is much easier to follow what's happening.